tests/system/Pager/PagerTest.php

```diff
@@ -103,6 +103,38 @@ public function testStoreAndHasMoreCanBeFalse()
 		$this->assertFalse($this->pager->hasMore('foo'));
 	}
 
+	public function testStoreWithQueries()
+	{
+		$_GET['page'] = 3;
+		$_GET['foo']  = 'bar';
+
+		$this->pager->store('default', 3, 25, 100);
+
+		$this->assertEquals('http://example.com?page=2&foo=bar', $this->pager->getPreviousPageURI());
+		$this->assertEquals('http://example.com?page=4&foo=bar', $this->pager->getNextPageURI());
+		$this->assertEquals('http://example.com?page=5&foo=bar', $this->pager->getPageURI(5));
+		$this->assertEquals(
+			'http://example.com?foo=bar&page=5',
+			$this->pager->only(['foo'])->getPageURI(5)
+		);
+	}
+
+	public function testStoreWithSegments()
+	{
+		$_GET['page'] = 3;
+		$_GET['foo']  = 'bar';
+
+		$this->pager->store('default', 3, 25, 100, 2);
```

Comment:
Noted this test also pass if the last param is 1.

Investigating, I just got here:

https://github.com/codeigniter4/CodeIgniter4/blob/cd8eba1fa72c45c199b5f0d3e91e2fdac91295b6/system/HTTP/URI.php#L494-L509

The `$number -= 1;` makes 1 or 2 acepptable.

Is this OK?

Reply:
See #1495

Reply:
Hmm. No that shouldn't work. The `number -= 1` though should just get it so that a user can think in 1-based arrays, but we're dealing with 0 based arrays. However, the +1 on line 500 looks weird to me.